common/test/java/com/couchbase/lite/internal/core/C4DatabaseTest.kt

```diff
@@ -0,0 +1,728 @@
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        createRev(targetDb, "doc001", REV_ID_1, fleeceBody)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+        try {
+            C4Database.copyDb(
+                srcDbPath!!,
+                dstParentDirPath,
+                dbName,
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.POSIX, ex.domain)
+            assertEquals(POSIX_EEXIST, ex.code)
+        }
+        targetDb = C4Database.getDatabase(
+            dstParentDirPath,
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+    }
+
+
+    // - Scopes and Collections
+
+    //  Test create 1 scope and 1 collection
+    @Test
+    fun testCreateScopeAndCollection() {
+        c4Database.addCollection("test_scope", "test_coll")
+        assertEquals(2, c4Database.scopeNames.size)
+        assertTrue(c4Database.hasScope("test_scope"))
+        assertEquals(setOf("_default", "test_scope"), c4Database.scopeNames)
+
+        assertEquals(1, c4Database.getCollectionNames("test_scope").size)
+        assertEquals(setOf("test_coll"), c4Database.getCollectionNames("test_scope"))
+    }
+
+    // Test create multiple collections in a scope
+    @Test
+    fun testCreateMultipleCollections() {
+        c4Database.addCollection("test_scope", "test_coll_1")
+        c4Database.addCollection("test_scope", "test_coll_2")
+
+        assertEquals(2, c4Database.getCollectionNames("test_scope").size)
+        assertEquals(setOf("test_coll_1", "test_coll_2"), c4Database.getCollectionNames("test_scope"))
+    }
+
+    @Test
+    fun testDeleteDefaultCollection() {
+        c4Database.deleteCollection(Scope.DEFAULT_NAME, Collection.DEFAULT_NAME)
+        assertNull(c4Database.defaultCollection)
+    }
+
+    // After deleting a collection, we can no longer get that collection (make sure test doesn't crash)
+    @Test
+    fun testDeleteCollection() {
+        val coll = c4Database.addCollection("test_scope", "test_coll")
+        c4Database.deleteCollection("test_scope", "test_coll")
+
+        //getting a non-existent collection should return null, not an exception
+        assertNull(c4Database.getCollection("test_scope", "test_coll"))
+
+        assertEquals(0, coll.documentCount)
+        assertThrows(LiteCoreException::class.java) {
+            coll.createDocument("1", null, 0)
+        }
+    }
+
+    // Close Database and collection operations will throw error
+    @Test
+    fun testCollectionOnClosedDB() {
+        val col = c4Database.getCollection(Scope.DEFAULT_NAME, Collection.DEFAULT_NAME)
+        c4Database.closeDb()
+        c4Database = null
```

Comment:
This might need a comment.... It is here to prevent the @After method from trying to close it again....